tests/Unit/PointwiseFunctions/Xcts/Test_AdmMomentum.cpp

```diff
@@ -128,39 +136,57 @@ void test_infinite_surface_integral(const double distance, const double mass,
         flat_face_normal.get(d) /= get(face_normal_magnitude);
       }
 
-      // Evaluate surface integral
-      const auto surface_integrand =
+      // Evaluate linear momentum surface integral
+      const auto linear_momentum_surface_integrand =
           Xcts::adm_linear_momentum_surface_integrand(
               conformal_factor, inv_spatial_metric, inv_extrinsic_curvature,
               trace_extrinsic_curvature);
-      const auto contracted_integrand = tenex::evaluate<ti::I>(
-          surface_integrand(ti::I, ti::J) * flat_face_normal(ti::j));
+      const auto linear_momentum_contracted_integrand = tenex::evaluate<ti::I>(
+          linear_momentum_surface_integrand(ti::I, ti::J) *
+          flat_face_normal(ti::j));
       for (int I = 0; I < 3; I++) {
-        surface_integral.get(I) += definite_integral(
-            contracted_integrand.get(I) * get(flat_area_element), face_mesh);
+        linear_momentum.get(I) +=
+            definite_integral(linear_momentum_contracted_integrand.get(I) *
+                                  get(flat_area_element),
+                              face_mesh);
       }
+
+      // Evaluate angular momentum surface integral
+      const auto angular_momentum_surface_integrand =
+          Xcts::adm_angular_momentum_z_surface_integrand(
+              linear_momentum_surface_integrand, inertial_coords);
+      const auto angular_momentum_contracted_integrand = tenex::evaluate(
+          angular_momentum_surface_integrand(ti::I) * flat_face_normal(ti::i));
+      angular_momentum_z.get() += definite_integral(
+          get(angular_momentum_contracted_integrand) * get(flat_area_element),
+          face_mesh);
     }
   }
 
-  // Check result
+  // Check results
   auto custom_approx = Approx::custom().epsilon(10. / distance).scale(1.0);
   const double lorentz_factor = 1. / sqrt(1. - square(boost_speed));
-  CHECK(get<0>(surface_integral) == custom_approx(0.));
-  CHECK(get<1>(surface_integral) == custom_approx(0.));
-  CHECK(get<2>(surface_integral) ==
+  CHECK(get<0>(linear_momentum) == custom_approx(0.));
+  CHECK(get<1>(linear_momentum) ==
         custom_approx(lorentz_factor * mass * boost_speed));
+  CHECK(get<2>(linear_momentum) == custom_approx(0.));
+  CHECK(get(angular_momentum_z) ==
+        custom_approx(x_shift * lorentz_factor * mass * boost_speed));
 }
 
 template <typename Solution>
 void test_infinite_volume_integral(const double distance, const double mass,
                                    const double horizon_radius,
                                    const double boost_speed,
                                    const Solution& solution) {
+  // Define x-shift in the domain
+  const double x_shift = 1.;
+
   // Set up domain
   const size_t h_refinement = 1;
   const size_t p_refinement = 6;
   const domain::creators::Sphere shell{
-      /* inner_radius */ 2 * horizon_radius,
+      /* inner_radius */ horizon_radius + x_shift,
```

Comment:
same here

Reply:
Same as above. Line 230 is responsible for shifting the coordinates.